```
FAILED: test/unit/test_log.p/test_log.c.o 
gcc-12 -Itest/unit/test_log.p -I. -I.. -Ilibrz -I../librz -Ilibrz/include -I../librz/include -Itest/unit -I../test/unit -Ilibrz/util/sdb/src -I../librz/util/sdb/src -Ilibrz/util/sdb -I../librz/util/sdb -I../librz/asm/arch/include -I../librz/asm/arch -I../librz/asm/arch/h8300 -I../librz/asm/arch/hexagon -I../librz/asm/arch/msp430 -I../librz/asm/arch/rsp -I../librz/asm/arch/mcore -I../librz/asm/arch/v850 -I../librz/asm/arch/propeller -I../librz/asm/arch/ebc -I../librz/asm/arch/cr16 -I../librz/asm/arch/8051 -I../librz/asm/arch/v810 -I../librz/asm/arch/or1k -I../librz/analysis/arch/gb -I../librz/bin/format/elf -I../librz/bin/format/dmp -I../librz/bin/format/mdmp -I../librz/bin/format/pe -I../librz/bin/format -Ilibrz/core/cmd_descs -I../librz/core/cmd_descs -I../librz/type/parser -fdiagnostics-color=always -fsanitize=address,undefined -fno-omit-frame-pointer -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -O2 -g -Wimplicit-fallthrough=3 -DRZ_PLUGIN_INCORE=1 -D_GNU_SOURCE --std=gnu99 -Werror=sizeof-pointer-memaccess -fvisibility=hidden -DASAN=1 -DRZ_ASSERT_STDOUT=1 -Wno-cpp -MD -MQ test/unit/test_log.p/test_log.c.o -MF test/unit/test_log.p/test_log.c.o.d -o test/unit/test_log.p/test_log.c.o -c ../test/unit/test_log.c
In file included from /usr/include/string.h:535,
                 from ../librz/include/rz_types.h:202,
                 from ../librz/include/rz_util.h:7,
                 from ../test/unit/test_log.c:4:
In function ‘memset’,
    inlined from ‘large_check’ at ../test/unit/test_log.c:17:2:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:59:10: error: ‘__builtin_memset’ offset [0, 998] is out of the bounds [0, 0] [-Werror=array-bounds]
   59 |   return __builtin___memset_chk (__dest, __ch, __len,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   60 |                                  __glibc_objsize0 (__dest));
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

--- a/librz/core/cmd/cmd.c
+++ b/librz/core/cmd/cmd.c
@@ -3582,7 +3582,7 @@ static void get_help_wrong_cmd(RzCore *core, const char *cmdname) {
 	if (!help_msg) {
 		goto help_pra_err;
 	}
-	RZ_LOG_ERROR("core: %s", help_msg);
+	RZ_LOG_ERROR("%s", help_msg);
 	free(help_msg);
 help_pra_err:
 	rz_cmd_parsed_args_free(help_pra);
@@ -3646,19 +3646,19 @@ DEFINE_HANDLE_TS_FCN_AND_SYMBOL(arged_stmt) {
 	res = rz_cmd_call_parsed_args(state->core->rcmd, pr_args);
 	if (res == RZ_CMD_STATUS_WRONG_ARGS) {
 		const char *cmdname = rz_cmd_parsed_args_cmd(pr_args);
-		RZ_LOG_ERROR("core: Wrong number of arguments passed to `%s`, see its help with `%s?`\n\n", cmdname, cmdname);
+		RZ_LOG_ERROR("Wrong number of arguments passed to `%s`, see its help with `%s?`\n\n", cmdname, cmdname);
 		get_help_wrong_cmd(state->core, cmdname);
 	} else if (res == RZ_CMD_STATUS_NONEXISTINGCMD) {
 		const char *cmdname = rz_cmd_parsed_args_cmd(pr_args);
-		RZ_LOG_ERROR("core: Command '%s' does not exist.\n", cmdname);
+		RZ_LOG_ERROR("Command '%s' does not exist.\n", cmdname);
 		if (rz_str_endswith(cmdname, "?") && pr_args->argc > 1) {
-			RZ_LOG_ERROR("core: Did you want to see the help? Try `%s` without any argument.\n", cmdname);
+			RZ_LOG_ERROR("Did you want to see the help? Try `%s` without any argument.\n", cmdname);
 		} else {
 			// Let's try to find the first command/group in the ancestor chain
 			// that could provide some help
 			RzCmdDesc *hcd = rz_cmd_get_desc_best(state->core->rcmd, cmdname);
 			if (hcd) {
-				RZ_LOG_ERROR("core: Displaying the help of command '%s'.\n\n", hcd->name);
+				RZ_LOG_ERROR("Displaying the help of command '%s'.\n\n", hcd->name);
 				get_help_wrong_cmd(state->core, hcd->name);
 			}
 		}

--- a/librz/util/log.c
+++ b/librz/util/log.c
@@ -85,17 +85,21 @@ RZ_API void rz_vlog(const char *funcname, const char *filename,
 	// TODO: Colors
 
 	// Build output string with src info, and formatted output
-	char output_buf[LOG_OUTPUTBUF_SIZE] = ""; // Big buffer for building the output string
+	RzStrBuf sb;
+	rz_strbuf_init(&sb);
+
 	if (!tag) {
 		tag = RZ_BETWEEN(0, level, RZ_ARRAY_SIZE(level_tags) - 1) ? level_tags[level] : "";
 	}
-	int offset = snprintf(output_buf, LOG_OUTPUTBUF_SIZE, "%s: ", tag);
+	rz_strbuf_append(&sb, tag);
+	rz_strbuf_append(&sb, ": ");
 	if (cfg_logsrcinfo) {
-		offset += snprintf(output_buf + offset, LOG_OUTPUTBUF_SIZE - offset, "%s in %s:%i: ", funcname, filename, lineno);
+		rz_strbuf_appendf(&sb, "%s in %s:%i: ", funcname, filename, lineno);
 	}
-	vsnprintf(output_buf + offset, LOG_OUTPUTBUF_SIZE - offset, fmtstr, args);
+	rz_strbuf_vappendf(&sb, fmtstr, args);
 
 	// Actually print out the string with our callbacks
+	char *output_buf = rz_strbuf_drain_nofree(&sb);
 	if (log_cbs && rz_list_length(log_cbs) > 0) {
 		RzListIter *it;
 		RzLogCallback cb;

--- a/test/unit/meson.build
+++ b/test/unit/meson.build
@@ -68,6 +68,7 @@ if get_option('enable_tests')
     'itv',
     'json',
     'list',
+    'log',
     'lzma',
     'ovf',
     'pj',

--- a/test/unit/test_log.c
+++ b/test/unit/test_log.c
@@ -1,0 +1,49 @@
+// SPDX-FileCopyrightText: 2022 Riccardo Schirone <sirmy15@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_util.h>
+#include "minunit.h"
+
+static int small_check(const char *output, const char *funcname, const char *filename,
+	ut32 lineno, RzLogLevel level, const char *tag, const char *fmtstr, ...) {
+	mu_assert_streq(output, "ERROR: 3", "small check msg should be correct");
+	return 0;
+}
+
+static int large_check(const char *output, const char *funcname, const char *filename,
+	ut32 lineno, RzLogLevel level, const char *tag, const char *fmtstr, ...) {
+	char *exp_msg = RZ_NEWS0(char, 2000);
+	strcpy(exp_msg, "ERROR: ");
+	for (size_t i = 0; i < 999; i++) {
+		exp_msg[strlen("ERROR: ") + i] = 'A';
+	}
+	mu_assert_streq(output, exp_msg, "large check msg should be correct");
+	free(exp_msg);
+	return 0;
+}
+
+bool test_log_small(void) {
+	rz_log_del_callback((RzLogCallback)large_check);
+	rz_log_add_callback((RzLogCallback)small_check);
+	rz_log("func", "file", 1, RZ_LOGLVL_ERROR, NULL, "%d", 3);
+	mu_end;
+}
+
+bool test_log_large(void) {
+	rz_log_del_callback((RzLogCallback)small_check);
+	rz_log_add_callback((RzLogCallback)large_check);
+	char *buf = RZ_NEWS(char, 1000);
+	memset(buf, 0x41, 999);
+	buf[999] = '\0';
+	rz_log("func", "file", 1, RZ_LOGLVL_ERROR, NULL, "%s", buf);
+	free(buf);
+	mu_end;
+}
+
+bool all_tests() {
+	mu_run_test(test_log_small);
+	mu_run_test(test_log_large);
+	return tests_passed != tests_run;
+}
+
+mu_main(all_tests)

